services/src/main/java/org/keycloak/services/util/CookieHelper.java

```diff
@@ -91,8 +91,17 @@ public static void addCookie(String name, String value, String path, String doma
         addCookie(name, value, path, domain, comment, maxAge, secure, httpOnly, null, session);
     }
 
+    public static String getCookieValue(KeycloakSession session, String name) {
+        Map<String, Cookie> cookies = session.getContext().getRequestHeaders().getCookies();
+        Cookie cookie = cookies.get(name);
+        if (cookie == null) {
+            String legacy = name + LEGACY_COOKIE;
+            cookie = cookies.get(legacy);
+        }
+        return cookie != null ? cookie.getValue() : null;
+    }
```

Comment:
Perhaps it would be slightly cleaner to have a separate method to retrieve the cookie, and only keep the value extraction logic in this method?

```java
public static String getCookieValue(KeycloakSession session, String name) {
    Cookie cookie = getCookie(session, name);
    return cookie != null ? cookie.getValue() : null;
}

private static Cookie getCookie(KeycloakSession session, String name) {
    Map<String, Cookie> cookies = session.getContext().getRequestHeaders().getCookies();
    Cookie cookie = cookies.get(name);

    if (cookie != null) {
        return cookie;
    }

    return cookies.get(name + LEGACY_COOKIE);
}
```

Reply:
Not sure I see the point in doing that, that would only really be useful if there was more than one place that used `getCookie`.

In either case this will end up being refactored more when it's brought into the CookieProvider eventually ;)

Reply:
Yeah, feel free to leave this to the side. I am sure future refactoring will touch this code anyways.